tools/cjxl_ng_main.cc

```diff
@@ -0,0 +1,593 @@
+    // documentation. This differs from what the old docstring said.
+    "Resampling for extra channels. Default of -1 applies resampling only "
+    "for low quality. Value 1 does no downsampling (1x1), 2 does 2x2 "
+    "downsampling, 4 is for 4x4 downsampling, and 8 for 8x8 downsampling.");
+
+DEFINE_int32(
+    epf, -1,
+    "Edge preserving filter level, -1 to 3. "
+    "Value -1 means: default (encoder chooses), 0 to 3 set a strength.");
+
+DEFINE_int64(
+    center_x, -1,
+    // TODO(tfish): Clarify if this is really the comment we want here.
+    "Determines the horizontal position of center for the center-first "
+    "group order. The value -1 means 'use the middle of the image', "
+    // TODO(tfish): Clarify if encode.h has an off-by-one in the
+    // upper limit here.
+    "other values 0..(xsize-1) set this to a particular coordinate.");
+
+DEFINE_int64(center_y, -1,
+             // TODO(tfish): Clarify if this is really the comment we want here.
+             "Determines the vertical position of center for the center-first "
+             "group order. The value -1 means 'use the middle of the image', "
+             // TODO(tfish): Clarify if encode.h has an off-by-one in the
+             // upper limit here.
+             "other values 0..(ysize-1) set this to a particular coordinate.");
+
+DEFINE_int64(num_threads, 0,
+             // TODO(tfish): Sync with team about changed meaning of 0 -
+             // was: No multithreaded workers. Is: use default number.
+             "Number of worker threads (0 == use machine default).");
+
+DEFINE_int64(num_reps, 1,  // TODO(tfish): wire this up.
+                           // TODO(tfish): Clarify meaning of this docstring.
+                           // Is this simply for benchmarking?
+             "How many times to compress.");
+
+DEFINE_int32(photon_noise, 0,
+             // TODO(tfish): Discuss docstring change with team.
+             // Also: This now is an int, no longer a float.
+             "Adds noise to the image emulating photographic film noise. "
+             "The higher the given number, the grainier the image will be. "
+             "As an example, a value of 100 gives low noise whereas a value "
+             "of 3200 gives a lot of noise. The default value is 0.");
+
+DEFINE_double(
+    distance, 1.0,  // TODO(tfish): wire this up.
+    "Max. butteraugli distance, lower = higher quality. Range: 0 .. 25.\n"
+    "    0.0 = mathematically lossless. Default for already-lossy input "
+    "(JPEG/GIF).\n"
+    "    1.0 = visually lossless. Default for other input.\n"
+    "    Recommended range: 0.5 .. 3.0.");
+
+DEFINE_int64(target_size, 0,  // TODO(tfish): wire this up.
+             "Aim at file size of N bytes.\n"
+             "    Compresses to 1 % of the target size in ideal conditions.\n"
+             "    Runs the same algorithm as --target_bpp");
+
+DEFINE_string(target_bpp, "0",  // TODO(tfish): wire this up.
```

Comment:
other non-double here and below

Reply:
Done. When wiring this up, need to make sure that `-inf` is correctly parsed for `-quality`.